tools/m2kcli/commands/digital/digital.cpp

```diff
@@ -0,0 +1,371 @@
+					 "  -h, --help            show this help message and exit\n"
+					 "  -q, --quiet           return result only\n"
+					 "  -c, --capture buffer_size=<size> [format=<type>]\n"
+					 "                        print a specific number of samples\n"
+					 "                        format - {csv | binary}; default csv\n"
+					 "  -9, --generate channel=<index>,... cyclic=<value> [format=<type>] [file=<path>]\n"
+					 "                        generate a signal\n"
+					 "                        by default the samples are read from stdin\n"
+					 "                        after writing all samples press CTRL+D(Unix) or CTRL+Z(Windows)\n"
+					 "                        one channel: channel=<index>\n"
+					 "                        many channels: channel=<index>,<index>,<index>...\n"
+					 "                        channel - {0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15}\n"
+					 "                        cyclic - 0 (disable)\n"
+					 "                               - 1 (enable)\n"
+					 "                        format - {csv | binary}; default csv\n"
+					 "                        file - the path of the file containing the samples\n"
+					 "  -g, --get [<attribute>...]\n"
+					 "                        return the value of the specified global attributes\n"
+					 "                        attribute:\n"
+					 "                            sampling_frequency_in\n"
+					 "                            sampling_frequency_out\n"
+					 "                            trigger_delay\n"
+					 "                            trigger_mode\n"
+					 "                            all\n"
+					 "  -G, --get-channel channel=<index> [<attribute> ...]\n"
+					 "                        return the value of the attributes corresponding to the given channel\n"
+					 "                        attributes:\n"
+					 "                            value\n"
+					 "                            output_mode\n"
+					 "                            triggerCondition\n"
+					 "                            all\n"
+					 "  -s, --set [<attribute>=<value>...]\n"
+					 "                        set the value of the specified global attributes\n"
+					 "                        attribute:\n"
+					 "                            sampling_frequency_in - double\n"
+					 "                            sampling_frequency_out - double\n"
+					 "                            trigger_delay - int\n"
+					 "                            trigger_mode - {or | and}\n"
+					 "  -S, --set-channel channel=<index> [<attribute>=<value> ...]\n"
+					 "                        set the value of the specified attributes corresponding to the given channel\n"
+					 "                        attributes: \n"
+					 "                            value - {0 | 1}\n"
+					 "                            output_mode - {open_drain | push_pull}\n"
+					 "                            trigger_digital_condition - {raising_edge | falling_edge | low_level | high_level | any_edge | no_trigger}\n";
```

Comment:
Same issue as before (in the analog_in side) . Should be "rising_edge".
Also, in the "--get-channel" part, we use "triggerCondition". Here we use "trigger_digital_condition".

Reply:
The help message was corrected in commit a61547e and the crashing was fixed in commit 52b4501.